### ISSUE TITLE
v5.10.0 util.c and pp_sys.c fail to compile under Mac OS X

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9080]
Migrated from [rt.perl.org#46725](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=46725) (status was 'resolved')

Searchable as RT46725$

**Comment by p5pRT** [comment]
### From @trwyant

This is a bug report for perl from wyant@&#8203;cpan\.org\,
generated with the help of perlbug 1\.36 running under perl 5\.10\.0\.




---
The 'make' step for Perl 5\.10\.0 DEVEL32174 fails to compile util\.c
under Mac OS X 10\.3\.9 with the following error&#8203;:

\`sh  cflags "optimize='\-Os'" util\.o\`  util\.c
&nbsp;          CCCMD =  cc \-DPERL\_CORE \-c \-DPERL\_PATCHNUM=32174 \-fno\-common \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe \-I/usr/local/include  \-std=c89 \-Os \-Wall \-ansi   \-W \-Wendif\-labels 
In file included from /usr/include/libkern/OSByteOrder\.h&#8203;:35\,
&nbsp;                 from /usr/include/mach/ndr\.h&#8203;:93\,
&nbsp;                 from /usr/include/mach/clock\_priv\.h&#8203;:7\,
&nbsp;                 from /usr/include/mach/mach\_interface\.h&#8203;:37\,
&nbsp;                 from /usr/include/mach/mach\.h&#8203;:62\,
&nbsp;                 from /usr/include/sys/event\.h&#8203;:154\,
&nbsp;                 from /usr/include/sys/select\.h&#8203;:71\,
&nbsp;                 from util\.c&#8203;:44&#8203;:
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:42&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:57&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:72&#8203;: error&#8203;: syntax error before "uint64\_t"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:93&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:107&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:121&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:143&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:153&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/ppc/OSByteOrder\.h&#8203;:163&#8203;: error&#8203;: syntax error before "uint64\_t"
In file included from /usr/include/mach/ndr\.h&#8203;:93\,
&nbsp;                 from /usr/include/mach/clock\_priv\.h&#8203;:7\,
&nbsp;                 from /usr/include/mach/mach\_interface\.h&#8203;:37\,
&nbsp;                 from /usr/include/mach/mach\.h&#8203;:62\,
&nbsp;                 from /usr/include/sys/event\.h&#8203;:154\,
&nbsp;                 from /usr/include/sys/select\.h&#8203;:71\,
&nbsp;                 from util\.c&#8203;:44&#8203;:
/usr/include/libkern/OSByteOrder\.h&#8203;:49&#8203;: error&#8203;: syntax error before "int32\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:105&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:115&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:125&#8203;: error&#8203;: syntax error before "uint64\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:137&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:148&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:159&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:172&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:182&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:192&#8203;: error&#8203;: syntax error before "uint64\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:204&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:215&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:226&#8203;: error&#8203;: syntax error before "void"
/usr/include/libkern/OSByteOrder\.h&#8203;:245&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:254&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:263&#8203;: error&#8203;: syntax error before "uint64\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:292&#8203;: error&#8203;: syntax error before "uint16\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:301&#8203;: error&#8203;: syntax error before "uint32\_t"
/usr/include/libkern/OSByteOrder\.h&#8203;:310&#8203;: error&#8203;: syntax error before "uint64\_t"
make&#8203;: \*\*\* \[util\.o\] Error 1

The problem is that the various OSByteOrder\.h files define \(by default\)
inline functions\, which are disallowed both by \-ansi and \-std=c89\. The
workaround is to compile without \-ansi\, and either without the \-std
option\, or with \-std=c99\.

pp\_sys\.c fails in the same way\, because it \(eventually\) includes the same
header\.

This may be a duplicate \- I tried to send a report on this on October 19\,
\(on DEVEL32147\) but I could never find it in rt\.perl\.org/perlbug\, meaning
either that I didn't send it correctly or didn't look it up correctly\.

I do not know what the favored response is\. A look at the Changes file
says that you dropped these two options under Cygwin\, but it was not
immediately obvious to me how to make this happen under Mac OS X\.

Note that this is \_NOT\_ the current Mac OS X by any means; the current is
10\.4\.10\, and is about to be 10\.5\. The output of 'uname \-a' is appended
for your edification and/or amusement\. Thanks\.

$ uname \-a
Darwin Wyants\-White\-iMac\.local 7\.9\.0 Darwin Kernel Version 7\.9\.0&#8203;: Wed Mar 30 20&#8203;:11&#8203;:17 PST 2005; root&#8203;:xnu/xnu\-517\.12\.7\.obj~1/RELEASE\_PPC  Power Macintosh powerpc


---


---
Flags&#8203;:
&nbsp;    category=install
&nbsp;    severity=low


---
Site configuration information for perl 5\.10\.0&#8203;:

Configured by tom at Mon Oct 22 18&#8203;:46&#8203;:15 EDT 2007\.

Summary of my perl5 \(revision 5 version 10 subversion 0 patch 32174\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=darwin\, osvers=7\.9\.0\, archname=darwin\-thread\-multi\-2level
&nbsp;    uname='darwin wyants\-white\-imac\.local 7\.9\.0 darwin kernel version 7\.9\.0&#8203;: wed mar 30 20&#8203;:11&#8203;:17 pst 2005; root&#8203;:xnuxnu\-517\.12\.7\.obj~1release\_ppc power macintosh powerpc '
&nbsp;    config\_args=''
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=undef\, use64bitall=undef\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-fno\-common \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe \-I/usr/local/include'\,
&nbsp;    optimize='\-Os'\,
&nbsp;    cppflags='\-no\-cpp\-precomp \-fno\-common \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='3\.3 20030304 \(Apple Computer\, Inc\. build 1671\)'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='env MACOSX\_DEPLOYMENT\_TARGET=10\.3 cc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib
&nbsp;    libs=\-ldbm \-ldl \-lm \-lc
&nbsp;    perllibs=\-ldl \-lm \-lc
&nbsp;    libc=/usr/lib/libc\.dylib\, so=dylib\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dyld\.xs\, dlext=bundle\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags=' \-bundle \-undefined dynamic\_lookup \-L/usr/local/lib'

Locally applied patches&#8203;:
&nbsp;    DEVEL



---
@&#8203;INC for perl 5\.10\.0&#8203;:
&nbsp;    /sw/lib/perl5
&nbsp;    /sw/lib/perl5/darwin
&nbsp;    /usr/local/test/perl/5\.10\.0/lib/5\.10\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/test/perl/5\.10\.0/lib/5\.10\.0
&nbsp;    /usr/local/test/perl/5\.10\.0/lib/site\_perl/5\.10\.0/darwin\-thread\-multi\-2level
&nbsp;    /usr/local/test/perl/5\.10\.0/lib/site\_perl/5\.10\.0
&nbsp;    \.



---
Environment for perl 5\.10\.0&#8203;:
&nbsp;    DYLD\_LIBRARY\_PATH \(unset\)
&nbsp;    HOME=/Users/tom
&nbsp;    LANG \(unset\)
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/sw/bin&#8203;:/sw/sbin&#8203;:/bin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/usr/sbin&#8203;:/Developer/Tools&#8203;:/Users/tom/Code/Tools&#8203;:/usr/X11R6/bin
&nbsp;    PERL5LIB=/sw/lib/perl5&#8203;:/sw/lib/perl5/darwin
&nbsp;    PERLPKGOPT=\-noppm
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 22 Oct 2007\, Tom Wyant wrote&#8203;:

> \# New Ticket Created by  Tom Wyant 
> \# Please include the string&#8203;:  \[perl \#46725\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=46725 >
> 

> The 'make' step for Perl 5\.10\.0 DEVEL32174 fails to compile util\.c
> under Mac OS X 10\.3\.9 with the following error&#8203;:
> 
> \`sh  cflags "optimize='\-Os'" util\.o\`  util\.c
>           CCCMD = cc \-DPERL\_CORE \-c \-DPERL\_PATCHNUM=32174 \-fno\-common 
>               \-DPERL\_DARWIN \-no\-cpp\-precomp \-fno\-strict\-aliasing \-pipe 
>               \-I/usr/local/include \-std=c89 \-Os \-Wall \-ansi \-W 
>               \-Wendif\-labels
> In file included from /usr/include/libkern/OSByteOrder\.h&#8203;:35\,
>                  from /usr/include/mach/ndr\.h&#8203;:93\,
>                  from /usr/include/mach/clock\_priv\.h&#8203;:7\,
>                  from /usr/include/mach/mach\_interface\.h&#8203;:37\,
>                  from /usr/include/mach/mach\.h&#8203;:62\,
>                  from /usr/include/sys/event\.h&#8203;:154\,
>                  from /usr/include/sys/select\.h&#8203;:71\,
>                  from util\.c&#8203;:44&#8203;:
> /usr/include/libkern/ppc/OSByteOrder\.h&#8203;:42&#8203;: error&#8203;: syntax error before "uint16\_t"
> /usr/include/libkern/ppc/OSByteOrder\.h&#8203;:57&#8203;: error&#8203;: syntax error before "uint32\_t"
> /usr/include/libkern/ppc/OSByteOrder\.h&#8203;:72&#8203;: error&#8203;: syntax error before "uint64\_t"

\[ etc\. \]

> The problem is that the various OSByteOrder\.h files define \(by default\)
> inline functions\, which are disallowed both by \-ansi and \-std=c89\. The
> workaround is to compile without \-ansi\, and either without the \-std
> option\, or with \-std=c99\.

Good analysis\.  The \-std=c89 and \-ansi flags are added by cflags\.SH after 
Configure time\.  cflags\.SH builds a "representative" C file that \#includes 
"perl\.h" and a whole bunch of system files\, and checks if it can be 
compiled with the flags \-std=c89 and \-ansi \(among others\)\.

That "representative" file already tries to include \<sys/select\.h>\, but 
because of a typo\, it doesn't\.  

This patch fixes a series of similar typos\.  It ought to fix it up&#8203;:

<details><summary>Inline Patch</summary>

```diff
diff -r -u perl-current/cflags.SH perl-andy/cflags.SH
--- perl-current/cflags.SH	2007-10-22 17:43:41.000000000 -0400
+++ perl-andy/cflags.SH	2007-10-23 08:28:18.000000000 -0400
@@ -66,31 +66,31 @@
 #ifdef I_UNISTD
 #include <unistd.h>
 #endif
-#ifdef I_SYSTYPES
+#ifdef I_SYS_TYPES
 #include <sys/types.h>
 #endif
-#ifdef I_SYSPARAM
+#ifdef I_SYS_PARAM
 #include <sys/param.h>
 #endif
-#ifdef I_SYSRESOURCE
+#ifdef I_SYS_RESOURCE
 #include <sys/resource.h>
 #endif
-#ifdef I_SYSSELECT
+#ifdef I_SYS_SELECT
 #include <sys/select.h>
 #endif
 #if defined(HAS_SOCKET) && !defined(VMS) && !defined(WIN32) /* See perl.h. */
 #include <sys/socket.h>
 #endif
-#ifdef I_SYSSTAT
+#ifdef I_SYS_STAT
 #include <sys/stat.h>
 #endif
-#ifdef I_SYSTIME
+#ifdef I_SYS_TIME
 #include <sys/time.h>
 #endif
-#ifdef I_SYSTIMES
+#ifdef I_SYS_TIMES
 #include <sys/times.h>
 #endif
-#ifdef I_SYSWAIT
+#ifdef I_SYS_WAIT
 #include <sys/wait.h>
 #endif
 /* The gcc -ansi can cause a lot of noise in Solaris because of:
@@ -139,7 +139,7 @@
        case " $ccflags " in
        *" $opt "*) ;; # Skip if already there.
        *) rm -f _cflags$_exe
-          case "`$cc $cflags $warn $opt _cflags.c -o _cflags$_exe 2>&1`" in
+          case "`$cc $cflags $warn $stdflags $opt _cflags.c -o _cflags$_exe 2>&1`" in
           *"unrecognized"*) ;;
           *"implicit declaration"*) ;; # Was something useful hidden?
           *"Invalid"*) ;;

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Oct 23\, 2007 at 08&#8203;:54&#8203;:51AM \-0400\, Andy Dougherty wrote&#8203;:
> This patch fixes a series of similar typos\.  It ought to fix it up&#8203;:

Thanks\, applied \(change 32181\)

Nicholas Clark

**Comment by p5pRT** [comment]
### From @trwyant

On Tue Oct 23 06&#8203;:32&#8203;:19 2007\, nicholas wrote&#8203;:
> On Tue\, Oct 23\, 2007 at 08&#8203;:54&#8203;:51AM \-0400\, Andy Dougherty wrote&#8203;:
> > This patch fixes a series of similar typos\.  It ought to fix it up&#8203;:
> 
> Thanks\, applied \(change 32181\)
> 
> Nicholas Clark
> 

Confirmed\. With 32181\, 'make' runs without intervention\, and 'make test'
without error\. Thank you very much\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
